client/signup/steps/difm-page-picker/index.tsx

```diff
@@ -0,0 +1,226 @@
+	}
+`;
+
+const GridCellContainer = styled.div< { isClickDisabled: boolean; isSelected: boolean } >`
+	cursor: ${ ( { isClickDisabled, isSelected } ) =>
+		! isSelected && isClickDisabled ? 'default' : 'pointer' };
+	opacity: ${ ( { isSelected, isClickDisabled } ) =>
+		! isSelected && isClickDisabled ? '0.4' : '1' };
+	min-width: 223px;
+	border-radius: 4px;
+	position: relative;
+	width: 100%;
+	position: relative;
+	display: flex;
+	flex-direction: column;
+	align-items: center;
+`;
+
+const CellLabelContainer = styled.div`
+	margin: 14px 0;
+	text-align: left;
+	display: flex;
+
+	justify-content: center;
+	& > div {
+		margin-right: 8px;
+	}
+	width: 100%;
+	@media ( min-width: 960px ) {
+		justify-content: left;
+	}
+`;
+
+const PopularContainer = styled.div`
+	width: 75px;
+	height: 25px;
+	left: 699px;
+	top: 352px;
+	background: #b8e6bf;
+	opacity: 0.64;
+	border-radius: 4px;
+	text-align: center;
+	& > div {
+		color: #00450c;
+		font-weight: 500;
+		text-align: center;
+	}
+`;
+
+interface PageCellType {
+	pageId: string;
+	selectedPages: string[];
+	onClick: ( pageId: string ) => void;
+	popular?: boolean;
+}
+
+function PageCell( { pageId, popular, selectedPages, onClick }: PageCellType ) {
+	const translate = useTranslate();
+	const selectedIndex = selectedPages.indexOf( pageId );
+	const totalSelections = selectedPages.length;
+	const isSelected = Boolean( selectedIndex > -1 );
+	const isDisabled = selectedIndex === -1 && totalSelections >= PAGE_LIMIT;
+	const title = useTranslatedPageTitles( pageId );
+
+	return (
+		<GridCellContainer
+			onClick={ () => onClick( pageId ) }
+			isSelected={ isSelected }
+			isClickDisabled={ isDisabled }
+		>
+			<BrowserView
+				pageId={ pageId }
+				isClickDisabled={ isDisabled }
+				isSelected={ isSelected }
+				selectedIndex={ selectedIndex >= 0 ? selectedIndex : -1 }
+			/>
+			<CellLabelContainer>
+				<div>{ title }</div>
+				{ popular ? (
+					<PopularContainer>
+						<div>{ translate( 'Popular' ) }</div>
+					</PopularContainer>
+				) : null }
+			</CellLabelContainer>
+		</GridCellContainer>
+	);
+}
+
+function PageSelector() {
+	const [ selectedPages, setSelectedPages ] = useState< string[] >( [ HOME_PAGE, ABOUT_PAGE ] );
+
+	const onPageClick = ( pageId: string ) => {
+		const foundIndex = selectedPages.indexOf( pageId );
+		if ( pageId !== HOME_PAGE )
+			if ( foundIndex > -1 ) {
+				// The home page cannot be touched
+				const tempArray = [ ...selectedPages ];
+				tempArray.splice( foundIndex, 1 );
+				setSelectedPages( tempArray );
+			} else if ( selectedPages.length !== PAGE_LIMIT ) {
+				setSelectedPages( [ ...selectedPages, pageId ] );
+			}
+	};
+
+	return (
+		<>
+			<PageGrid>
+				<PageCell
+					popular
+					pageId={ HOME_PAGE }
+					selectedPages={ selectedPages }
+					onClick={ onPageClick }
+				/>
+				<PageCell
+					popular
+					pageId={ ABOUT_PAGE }
+					selectedPages={ selectedPages }
+					onClick={ onPageClick }
+				/>
```

Comment:
I know that you are against looping based on your comment here: https://github.com/Automattic/wp-calypso/pull/61896/files#r827613233 😄 

However, I wonder how well this can scale. For example, what if we add more (~20) pages here or fetch this from the server? This code is already very verbose, and we are repeating almost all props here. You have already encapsulated logic into the `PageCell` component, so why not use a loop here? We can also simplify this by replacing the `selectedPages` prop with `isSelected` & `isDisabled`. This way, the display logic (how to render) of `PageCell` will be separated from the "what to render" logic.

LMK what you think. 🙂

Reply:
This can scale beautifully with 20 elements far better than putting it into a separate JSON structure  ! 

> This code is already very verbose, and we are repeating almost all props here. 

If we move this to a data structure aren't we also repeating the very same props there, but now you have the added complexity of having to go over this structure  with a loop?

>We can also simplify this by replacing the selectedPages prop with isSelected & isDisabled

This will be a step backwards in encapsulating the view logic for the page cell component.

> This way, the display logic (how to render) of PageCell will be separated from the "what to render" logic.

Don't see how this can be helpful. When reasoning about this component I looked at what decisions makes high cohesion. The decision to show and hide something within a given component is that component's responsibility. We should provide the relevant props to that component to make that decision and hide away that complexity from the parent component. Delegate as much as possible specially when the logic can get a bit convoluted.